hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -3087,6 +3139,28 @@ static void pgraph_bind_shaders(PGRAPHState *pg)
 
     glUseProgram(pg->shader_binding->gl_program);
 
+    /* Clipping regions */
+    for (i = 0; i < state.psh.window_clip_count; i++) {
+        char name[32];
+
+        uint32_t x = pg->regs[NV_PGRAPH_WINDOWCLIPX0 + i * 4];
+        GLuint xmin = GET_MASK(x, NV_PGRAPH_WINDOWCLIPX0_XMIN);
+        GLuint xmax = GET_MASK(x, NV_PGRAPH_WINDOWCLIPX0_XMAX);
+        uint32_t y = pg->regs[NV_PGRAPH_WINDOWCLIPY0 + i * 4];
+        GLuint ymin = pg->surface_shape.clip_height - 1 -
+                      GET_MASK(y, NV_PGRAPH_WINDOWCLIPX0_XMIN);
+        GLuint ymax = pg->surface_shape.clip_height - 1 -
+                      GET_MASK(y, NV_PGRAPH_WINDOWCLIPX0_XMAX);
+
+        snprintf(name, sizeof(name), "clipRegion[%d]", i);
+        GLint loc = glGetUniformLocation(pg->shader_binding->gl_program, name);
```

Comment:
This uniform location should be cached, as done for other shader uniforms (I believe?).
This can have quite a large performance impact.

Reply:
Done